executor/src/witgen/generator.rs

```diff
@@ -148,9 +145,7 @@ where
                 break;
             }
         }
-        // Identity check failure on the first row is not fatal. We will proceed with
-        // "unknown", report zero and re-check the wrap-around against the zero values at the end.
-        if identity_failed && next_row != 0 {
+        if identity_failed {
```

Comment:
What's the effect of this change? Is `next_row` expected to never be 0?

Reply:
Yes, see https://github.com/powdr-org/powdr/pull/151/files#diff-06c2d788cbdf696f356c8bfb53645c75075218b7b65aa5456a76b33d9d122e87L121

Do you mean that it should be asserted here?

Reply:
No: before this change, as far as I understand a failed identity on the first row was somehow not an issue. 
I suspect this change is correct even outside the context of this PR, because since we introduced the `EvalResult` I can't see how we can recover from a failed identity (the message below should also be adjusted).